src/thing_physics.c

```diff
@@ -902,8 +903,13 @@ TbBool move_object_to_nearest_free_position(struct Thing *thing)
             if (!thing_in_wall_at_with_radius(thing, &pos, nav_radius))
             {
                 pos.z.val = get_thing_height_at(thing, &pos);
-                move_thing_in_map(thing, &pos);
-                return true;
+                if (thing_exists(thing))
```

Comment:
why would you first look for a spot and only then check if it exists, wouldn't it make more sense at start of func?